File: two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        h_map = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in h_map:
                return [h_map[diff], i]
            else:
                h_map[nums[i]] = i

File: cpp/1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> my_map;
        vector<int> res;

        for(int i = 0; i < nums.size(); i++){
            if(my_map.find(target - nums[i]) != my_map.end()){
                res.push_back(my_map[target - nums[i]]);
                res.push_back(i);
                return res;
            }
            else{
                my_map[nums[i]] = i;
            }
        }
        return res;
    }
};